### PR TITLE
Parse bank invoice url before displaying it

### PR DESCRIPTION
**Pull request by kaisermann**
#### What does this PR do? \*

The PaymentMethod component is also used on the `order-placed` page. If a user is not logged in and chose `boleto bancário` as the payment method, we should make the "open boleto" button redirect to the log-in page.

#### How to test it? \*

Go to:

https://storetheme.vtex.com/ and buy something while unlogged

After getting to the order-placed page, add a `&workspace=kiwi` to your url.

image

#### Related to / Depends on \*

Related to https://github.com/vtex-apps/order-placed/pull/129

### CODE REVIEW

File: react/PaymentMethod.tsx [hunk 1]
```diff
@@ -82,7 +83,10 @@ const PaymentMethod: FunctionComponent<Props & InjectedIntlProps> = ({
         </div>
         {isBankInvoice && payment.url && (
           <div className="mt5">
-            <ButtonLink to={payment.url} variation="primary" openNewWindow>
+            <ButtonLink
+              to={parseBankInvoiceUrl(payment.url)}
+              variation="primary"
+              openNewWindow>
```
Comment (victorhmp) [inline react/PaymentMethod.tsx:91]:
```suggestion
              openNewWindow
             >
```

File: react/utils/index.tsx [hunk 2]
```diff
@@ -151,3 +151,17 @@ export const orderSplitMessage = ({
...
   return intl.formatMessage(message, { deliveries, pickups, takeaways })
 }
+
+
+
+export function parseBankInvoiceUrl(url: string) {
+  const isEncrypted = Boolean(url.match(/(\*.\*.)+\*\w\*/g))
+  if (!isEncrypted) return url
+
+  const rootPath = window?.__RUNTIME__.rootPath ?? ''
```
Comment (victorhmp) [inline react/utils/index.tsx]: Why is this necessary? Wouldn't `useRuntime` work in this case?

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: react/PaymentMethod.tsx (hunk 1 above)
Comment (kaisermann) [reply react/PaymentMethod.tsx:91]: Will format the project in a future PR, it's waaaay outdated.

File: react/utils/index.tsx (hunk 2 above)
Comment (kaisermann) [reply react/utils/index.tsx]: I just copied it from `order-placed` :sad:

File: react/utils/index.tsx [hunk 3]
```diff
@@ -151,3 +151,17 @@ export const orderSplitMessage = ({
 
   return intl.formatMessage(message, { deliveries, pickups, takeaways })
 }
+
+
+
```
Comment (klzns) [inline react/utils/index.tsx]:
```suggestion
```

**Review by klzns — APPROVED** [review]

**Review by salesfelipe — APPROVED** [review]

**Review by victorhmp — APPROVED** [review]
